EnableIPythonDisplay.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Hooks IPython to the KernelCommunicator, so that it can send display
+/// messages to Jupyter.
+
+import Python
+
+// Workaround SR-7757.
+#if canImport(Darwin)
+import func Darwin.C.dlopen
+#elseif canImport(Glibc)
+import func Glibc.dlopen
+#else
+#error("Cannot import Darwin or Glibc!")
+#endif
+dlopen("libpython2.7.so", RTLD_NOW | RTLD_GLOBAL)
+
+enum IPythonDisplay {
+  static var socket: PythonObject = Python.None
+  static var shell: PythonObject = Python.None
+
+  static func enableIPythonDisplay() {
```

Comment:
I would also suggest defining methods, including helper methods as private methods in an `extension`. It is not very idiomatic to define many levels of nested functions (in here `bytes(_:)` is a 3-level nested function).

Reply:
done